packages/taco/src/conditions/schemas/sequential.ts

```diff
@@ -1,10 +1,56 @@
 import { z } from 'zod';
 
+import { ConditionProps } from '../condition';
 import { maxNestedDepth } from '../multi-condition';
 
 import { baseConditionSchema, plainStringSchema } from './common';
+import { CompoundConditionType } from './compound';
+import { IfThenElseConditionType } from './if-then-else';
 import { anyConditionSchema } from './utils';
 
+const getAllNestedConditionVariableNames = (
+  condition: ConditionProps,
+): string[] => {
+  const conditionVariables: string[] = [];
+  if (condition.conditionType === SequentialConditionType) {
+    for (const variable of condition.conditionVariables) {
+      conditionVariables.push(variable.varName);
+      conditionVariables.push(
+        ...getAllNestedConditionVariableNames(variable.condition),
+      );
+    }
+  } else if (condition.conditionType === IfThenElseConditionType) {
+    conditionVariables.push(
+      ...getAllNestedConditionVariableNames(condition.ifCondition),
+    );
+    conditionVariables.push(
+      ...getAllNestedConditionVariableNames(condition.thenCondition),
+    );
+    if (typeof condition.elseCondition !== 'boolean') {
+      conditionVariables.push(
+        ...getAllNestedConditionVariableNames(condition.elseCondition),
+      );
+    }
+  } else if (condition.conditionType === CompoundConditionType) {
+    for (const operand of condition.operands) {
+      conditionVariables.push(...getAllNestedConditionVariableNames(operand));
+    }
+  }
+  return conditionVariables;
+};
+
+const noDuplicateVarNames = (condition: ConditionProps): boolean => {
+  const allVarNames = getAllNestedConditionVariableNames(condition);
+  const seen = new Set();
+  for (const varName of allVarNames) {
+    if (seen.has(varName)) {
+      return false; // Duplicate variable name found
+    }
+    seen.add(varName);
+  }
+  return true; // No duplicates found
```

Comment:
I don't see any problem with this, but if you would like to do it in a more javascript-style way, you could try something like

```suggestion
  const duplicates = allVarNames.filter((item, index) => allVarNames.indexOf(item) !== index);
  return duplicates.length === 0;
```

Reply:
✔️ I'll take it - I think your suggestion is better